test/ConstructiveSolidGeometry/CSG_test.jl

```diff
@@ -42,73 +37,205 @@ end
+    @code_warntype cp1 in rot_box
+    @code_warntype cp1 in rot_stretched_box
+    @code_warntype cp1 in translated_box
+    @code_warntype cp1 in translated_rotated_box
+    @code_warntype cp1 in translated_rot_stretched_box
+end
+
+@btime cp1 in translated_rot_stretched_box
+
+
+rot = RotX(deg2rad(90))
+t = CartesianVector{T}(1, 0, 0 )
+sphere = Sphere(1.0)
+stretching = SVector(1.0, 3.0, 1.0)
+stretched_sphere = StretchedGeometry(sphere, stretching)
+rot_sphere = RotatedGeometry(sphere, RotMatrix(rot))
+translated_sphere = TranslatedGeometry(sphere, t)
+translated_rotated_sphere = TranslatedGeometry(rot_sphere, t)
+rot_stretched_sphere =  RotatedGeometry(stretched_sphere, RotMatrix(rot))
+translated_rot_stretched_sphere = TranslatedGeometry(rot_stretched_sphere, t)
+cp1 = CartesianPoint{T}(0, 0, 1)
+
+if code_warntype
+    @code_warntype cp1 in sphere
+    @code_warntype cp1 in stretched_sphere
+    @code_warntype cp1 in rot_sphere
+    @code_warntype cp1 in rot_stretched_sphere
+    @code_warntype cp1 in translated_sphere
+    @code_warntype cp1 in translated_rotated_sphere
+    @code_warntype cp1 in translated_rot_stretched_sphere
+end
+
+@btime cp1 in translated_rot_stretched_sphere
+
+
+rot = RotX(deg2rad(90))
+t = CartesianVector{T}(1, 0, 0 )
+hexagon = HexagonalPrism(0.5,1.0,-1.0,1.0)
+stretching = SVector(1.0, 3.0, 1.0)
+stretched_hexagon = StretchedGeometry(hexagon, stretching)
+rot_hexagon = RotatedGeometry(hexagon, RotMatrix(rot))
+translated_hexagon = TranslatedGeometry(hexagon, t)
+translated_rotated_hexagon = TranslatedGeometry(rot_hexagon, t)
+rot_stretched_hexagon =  RotatedGeometry(stretched_hexagon, RotMatrix(rot))
+translated_rot_stretched_hexagon = TranslatedGeometry(rot_stretched_hexagon, t)
 cp1 = CartesianPoint{T}(0, 0, 1)
 
-@code_warntype cp1 in tube
-@code_warntype cp1 in streched_tube
-@code_warntype cp1 in rot_tube
-@code_warntype cp1 in rot_streched_tube
-@code_warntype cp1 in translated_tube
-@code_warntype cp1 in translated_rotated_tube
-@code_warntype cp1 in translated_rot_streched_tube
+if code_warntype
+    @code_warntype cp1 in hexagon
+    @code_warntype cp1 in stretched_hexagon
+    @code_warntype cp1 in rot_hexagon
+    @code_warntype cp1 in rot_stretched_hexagon
+    @code_warntype cp1 in translated_hexagon
+    @code_warntype cp1 in translated_rotated_hexagon
+    @code_warntype cp1 in translated_rot_stretched_hexagon
+end
 
-@btime cp1 in translated_rot_streched_tube
+@btime cp1 in translated_rot_stretched_hexagon
 
 using Plots
 begin
     plts = []
     for p in [  
                 tube, 
-                streched_tube, 
+                stretched_tube, 
                 rot_tube, 
-                rot_streched_tube, 
+                rot_stretched_tube, 
                 translated_tube, 
                 translated_rotated_tube ,
-                translated_rot_streched_tube,
+                translated_rot_stretched_tube,
             ]
         xs = T[]
         ys = T[]
```

Comment:
Can you please remove my plotting and example lines and put this into a clean test file?